src/objects/core/zcl_abapgit_objects_activation.clas.abap

```diff
@@ -115,10 +115,12 @@ CLASS zcl_abapgit_objects_activation IMPLEMENTATION.
 
       CALL FUNCTION 'DD_MASS_ACT_C3'
         EXPORTING
-          ddmode         = 'O'
-          medium         = 'T' " transport order
-          device         = 'T' " saves to table DDRPH?
-          version        = 'M' " activate newest
+          ddmode         = 'T'         " mimic Activation during Transport
```

Comment:
ddmode = T does not seem right since objects are not created via transport (ie. no E071* entries, no import order, etc).

Reply:
DDMODE = 'O' + FRACT = abap_true somehow didn't and still doesn't work for me - maybe I have to start over with fresh head tomorrow. 

In fact, DDMODE = 'T' + INACTIVE = abap_true + DELALL = abap_false is thus far the only combo I've seen, which doesn't give any DDIC activation error for this case (there was another combo, which gave activation error only for the view).

My experiment with DDMODE = 'T' is doing what is called 'inactive activation' in one OSS Note (DDIC active, but DB objects are still in the previous state, if I understand correctly). What's still missing at the moment in the PR are the subsequent CTS 'transport independent' steps:

1) mass distribution of  DDIC changes, done by CTS JOB using PROG RDDDISTB (FUBA DD_MASS_DIST), and
2) "moving nametabs', done by CTS Job using PROG RDDMNTAB (FUBA TRINT_MOVE_NAMETAB),

out of table DDXTT.

Adding in those two SUBMITs _"seems"_ to do the trick (for this case), but all of this needs extensive testing, with more than a table and a view. Is there perhaps a suitable repo somewhere - with lots of  tricky, interdependent DDIC changes?

And, those FUBAs are obviously 2 more tricky and very risky dependencies on SAP code, which by right only SAP itself should be depending on (TRINT_MOVE_NAMETAB is fundamentally a C-call)... 

The reason I'm trying to mimic CTS is... mostly irrational trust :) (haven't seen DDIC activation/conversion problems in transports for many, many years; if there is a bug in CTS, SAP ought to fix it, right...).

Reply:
@mbtools I have reverted to ddmode = 'O' + frcact = abap_true; should work, as per your tests, on 7.50, presumably the initial delivery, or higher. I'd suggest to test the behavior when the table contains data (ddmode = 'G' + frcact = abap_true, which in 7.40 SP 23 activates https://github.com/JanisB-AT/Test_4496 branch1 to branch2 changes successfully, fails if the table contains data).

@larshp should I continue investing in the 'inactive activation (of DDIC)' (ddmode = 'T' + inactive = 'X' + delall = 'X') path? It's not entirely hopeless, I believe ("CTS does it" during import in the target system, after all), but it's difficult to judge beforehand, what am I actually getting into... 

To recapitulate: on 7.40 SP 23 that path would involve up to 3 (as the knowledge stands now...) additional steps, in order to get the TABL 'DB Object' and the DB itself up to speed with activated DDIC version:
- distribution of DDIC changes (using RDDDIS0L/RDDDISTB - FUBA DD_MASS_DIST)
- table conversion if there is data (possibly using something from RADBAT01)
- moving of nametabs (using RDDMNTAB - FUBA TRINT_MOVE_NAMETAB)

Should I still submit the current state of PR for review?